Very nice refactor. Build fail real:
```
======================================================================
FAIL: test_Broken_Extension_Chain (casexml.apps.phone.tests.test_extension_indexes.IndexTreeTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/corehq/util/test_utils.py", line 232, in inner
    return helper(*args, **kwargs)
  File "/mnt/corehq/util/test_utils.py", line 205, in __call__
    call_with_settings(fn_with_pre_and_post, run_config.settings, args, kwargs)
  File "/mnt/corehq/util/test_utils.py", line 220, in call_with_settings
    fn(*args, **kwargs)
  File "/mnt/corehq/util/test_utils.py", line 201, in fn_with_pre_and_post
    self.fn(*args, **kwargs)
  File "/mnt/corehq/util/test_utils.py", line 87, in wrapper
    return func(*args, **kw)
  File "corehq/ex-submodules/casexml/apps/phone/tests/test_extension_indexes.py", line 39, in test
    assert_user_doesnt_have_cases(self, self.user, undesired_cases)
  File "corehq/ex-submodules/casexml/apps/case/tests/util.py", line 123, in assert_user_doesnt_have_cases
    should_have=False, **kwargs)
  File "corehq/ex-submodules/casexml/apps/case/tests/util.py", line 193, in check_user_has_case
    matches = [check_block(case_block) for case_block in case_blocks]
  File "corehq/ex-submodules/casexml/apps/case/tests/util.py", line 185, in check_block
    "but shouldn't:%s" % (user.username, case_id, extra_info())
AssertionError: User 'syncguy' gets case 'b' but shouldn't:
<ns0:case xmlns:ns0="http://commcarehq.org/case/transaction/v2" case_id="b" date_modified="2016-06-10T01:08:32.959712Z" />
['<ns0:case xmlns:ns0="http://commcarehq.org/case/transaction/v2" case_id="a" date_modified="2016-06-10T01:08:31.794765Z" user_id="68e6e674fe69024180ece0380057023b"><ns0:create><ns0:case_type>mother</ns0:case_type><ns0:case_name /><ns0:owner_id>68e6e674fe69024180ece0380057023b</ns0:owner_id></ns0:create></ns0:case>', '<ns0:case xmlns:ns0="http://commcarehq.org/case/transaction/v2" case_id="c" date_modified="2016-06-10T01:08:31.794903Z" user_id="68e6e674fe69024180ece0380057023b"><ns0:create><ns0:case_type>mother</ns0:case_type><ns0:case_name /><ns0:owner_id>943737b1fff64ccd9cd91a53e7cda01a</ns0:owner_id></ns0:create><ns0:index><ns0:extension_1 case_type="default_related_case_type" relationship="extension">b</ns0:extension_1></ns0:index></ns0:case>', '<ns0:case xmlns:ns0="http://commcarehq.org/case/transaction/v2" case_id="b" date_modified="2016-06-10T01:08:31.795112Z" user_id="68e6e674fe69024180ece0380057023b"><ns0:create><ns0:case_type>mother</ns0:case_type><ns0:case_name /><ns0:owner_id>95cccf9c2b1942c8b5516c333debcf60</ns0:owner_id></ns0:create><ns0:index><ns0:extension_0 case_type="default_related_case_type" relationship="extension">a</ns0:extension_0></ns0:index><ns0:close /></ns0:case>']
-------------------- >> begin captured stdout << ---------------------
<function test at 0x7f25f6f13e60> failed with the following settings:
settings.TESTS_SHOULD_USE_SQL_BACKEND = False
```

--- a/corehq/ex-submodules/casexml/apps/phone/data_providers/case/clean_owners.py
+++ b/corehq/ex-submodules/casexml/apps/phone/data_providers/case/clean_owners.py
@@ -50,7 +50,6 @@ class CleanOwnerSyncPayload(object):
 
     def _get_next_case_batch(self):
         ids = pop_ids(self.case_ids_to_sync, chunk_size)
-        # TODO: see if we can avoid wrapping - serialization depends on it heavily for now
         return [
             case for case in self.case_accessor.get_cases(ids)
             if not case.is_deleted and case_needs_to_sync(case, last_sync_log=self.restore_state.last_sync_log)

--- a/corehq/ex-submodules/casexml/apps/phone/data_providers/case/clean_owners.py
+++ b/corehq/ex-submodules/casexml/apps/phone/data_providers/case/clean_owners.py
@@ -143,7 +143,7 @@ class CleanOwnerSyncPayload(object):
         self.restore_state.current_sync_log.closed_cases = self.closed_cases
 
     def purge_and_get_irrelevant_cases(self):
-        original_case_ids_on_phone = self.restore_state.current_sync_log.case_ids_on_phone
+        original_case_ids_on_phone = self.restore_state.current_sync_log.case_ids_on_phone.copy()
         self.restore_state.current_sync_log.purge_dependent_cases()
         purged_cases = original_case_ids_on_phone - self.restore_state.current_sync_log.case_ids_on_phone
         # don't sync purged cases that were never on the phone

--- a/corehq/ex-submodules/casexml/apps/phone/tests/test_extension_indexes.py
+++ b/corehq/ex-submodules/casexml/apps/phone/tests/test_extension_indexes.py
@@ -24,6 +24,10 @@ def get_test_file_json(filename):
     return json.loads(file_contents)
 
 
+def get_test_name(test_name):
+    return str("test_%s" % re.sub("\s", "_", test_name))
+
+
 @nottest
 def test_generator(test_name, skip=False):
     @softer_assert
@@ -37,6 +41,8 @@ def test_generator(test_name, skip=False):
         self.assertEqual(sync_log.case_ids_on_phone, set(desired_cases))
         assert_user_has_cases(self, self.user, desired_cases)
         assert_user_doesnt_have_cases(self, self.user, undesired_cases)
+
+    test.__name__ = get_test_name(test_name)
     return run_with_all_backends(test)
 
 
@@ -50,7 +56,7 @@ class TestSequenceMeta(type):
 
         for test in [(test['name'], test.get('skip', False)) for test in tests_to_run]:
             # Create a new testcase that the test runner is able to find
-            dict["test_%s" % re.sub("\s", "_", test[0])] = test_generator(test[0], test[1])
+            dict[get_test_name(test[0])] = test_generator(test[0], test[1])
 
         return type.__new__(mcs, name, bases, dict)
 

